src/App.tsx

```diff
@@ -8,37 +8,88 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getPostsData, getUsersData } from './api/posts';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User>();
+  const [selectedPost, setSelectedPost] = useState<Post>();
+  const [loader, setLoader] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<boolean>(false);
+  const [createNewComment, setCreateNewComment] = useState<boolean>(false);
+
+  useEffect(() => {
+    getUsersData()
+      .then(setUsers)
+      .catch(() => setErrorMessage(true));
+  }, []);
+
+  useEffect(() => {
+    if (selectedUser) {
+      setLoader(true);
+      setErrorMessage(false);
+
+      getPostsData(selectedUser.id)
+        .then(setUserPosts)
+        .catch(() => setErrorMessage(true))
+        .finally(() => setLoader(false));
+    }
+  }, [selectedUser]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                  setSelectedPost={setSelectedPost}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                <Loader />
+                {loader && <Loader />}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {errorMessage && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {selectedUser
+                  && !userPosts.length
+                  && !errorMessage
+                  && !loader && (
```

Comment:
Please fix it 
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/70009205/e475e026-9f85-4c49-baef-a1de6bef5750)